src/dmd/errors.d

```diff
@@ -161,6 +165,8 @@ final class StderrDiagnosticReporter : DiagnosticReporter
     private int warningCount_;
     private int deprecationCount_;
 
+  nothrow:
```

Comment:
The intention of `DiagnosticReporter` is to be used by a library to do whatever it wants.

Reply:
Exception handling makes RAII objects slow.

Reply:
I guess `nothrow` is acceptable, but I don't think we can add any other attributes, if you plan for that.